src/lib.rs

```diff
@@ -321,7 +322,22 @@ impl LightningNode {
         &self,
         invoice: String,
     ) -> std::result::Result<InvoiceDetails, DecodeInvoiceError> {
-        todo!()
+        match self.rt.block_on(parse(&invoice)) {
+            Ok(input_type) => match input_type {
+                InputType::Bolt11 {
```

Comment:
I've just tried it, and it doesn't help much because we still need to handle the other `InputType`s, meaning that we have to add a catch-all `_ => {}` term anyway.